actions/workflows/npm_publish_version.yaml

```diff
@@ -0,0 +1,43 @@
+---
+  chain:
+    -
+      name: "get_build_server"
+      ref: "linux.dig"
+      params:
+        hostname: "st2-build-slave-ubuntu.service.consul"
+        rand: true
+        count: 1
+      publish:
+        build_server: "{{get_build_server.result[0]}}"
+      on-success: "clone_repo"
+    -
+      name: "clone_repo"
+      ref: "st2cd.git_clone"
+      params:
+        hosts: "{{build_server}}"
+        repo: "{{repo}}"
+        branch: "{{branch}}"
+        target: "{{repo_target}}"
+      publish:
+        repodir: "{{clone_repo[build_server].stdout}}"
+      on-success: "npm_publish"
+    -
+      name: "npm_publish"
+      ref: "core.remote"
+      params:
+        hosts: "{{build_server}}"
+        cmd: "cd {{repodir}} && npm version {{update_type}} && npm publish"
```

Comment:
> I assume we want to increment the version ourselves since npm version is naive and doesn't know which version you want to increment (major / minor / patch).

Actually the ``update_type `` is exactly that i.e. we get to pick major/minor/patch and ``npm version`` assuming semantic versioning does the right thing. The has worked well for me manually so I simply replicated that approach.

> What is the safety if I try and tag a module with something I already have?

Don't tag manually - that my current process :)

> Also, is there an opportunity to leverage Git Tags here?

``npm version`` actually creates a tag matching the new version locally. So we are leveraging git tags except that ``npm version`` is performing the tagging so its not obvious. Btw, notice the ``git_commit_updates`` task in the workflow - thats where the tag gets pushed.